Trigger null-selection in SongSelect when the last beatmap is hidden

FreezyLemon opened the pull request:
I also removed the `SelectNext()` call since it's not needed for removing maps, so it shouldn't be necessary for hiding the last map of a set either (the set disappears for both actions, and it works without that method call in my testing).

Closes #1723 .

FreezyLemon committed:
Removed unnecessary SelectNext() call and change to dummy map when no items present

peppy commented on osu.Game/Screens/Select/BeatmapCarousel.cs:
```diff
@@ -512,7 +511,7 @@ void performMove(float y, float? startY = null)
             scrollableContent.Height = currentY;
 
-            if (selectedBeatmapSet != null && selectedBeatmapSet.State.Value != CarouselItemState.Selected)
+            if (!Items.Any() || selectedBeatmapSet != null && selectedBeatmapSet.State.Value != CarouselItemState.Selected)
```
won't this mean a beatmap is left selected if a search matches nothing?

peppy replied:
never mind, I read this back-to-front.

FreezyLemon committed:
Added tests for hiding diffs (and especially the last visible diff)

FreezyLemon commented:
This also works perfectly fine with #1726 . Just add the `!Items.Any() ||` in front of the `selectedBeatmapSet == null` similar to how it's done here and it works.

peppy committed:
Merge branch 'master' into fix-hiding-last-map

peppy committed:
Switch to a better way of deciding on null selection

This avoids `SelectionChanged` potentially being invoked multiple times after `selectedBeatmapSet` is already `null`.
